api/texts_api.py

```diff
@@ -0,0 +1,170 @@
+import django
+django.setup()
+from sefaria.utils.hebrew import hebrew_term
+from .api_errors import *
+from .helper import split_query_param_and_add_defaults
+from typing import List
+
+
+class VersionsParams():
+    """
+    an object for managing the versions params for TextsHandler
+    params can come from an API request or internal (sever side rendering)
+    lang is our code language. special values are:
+        source - for versions in the source language of the text
+        base - as source but with falling to the 'nearest' to source, or what we have defined as such
+    vtitle is the exact versionTitle. special values are:
+        no vtitle - the version with the max priority attr of the specified language
+        all - all versions of the specified language
+    representing_string is the original string that came from an API call
+    """
+
+    def __init__(self, lang: str, vtitle: str, representing_string=''):
+        self.lang = lang
+        self.vtitle = vtitle
+        self.representing_string = representing_string
+        if not self.representing_string:
+            self.representing_string = f'{self.lang}|{self.representing_string}'
+
+    def __eq__(self, other):
+        return isinstance(other, VersionsParams) and self.lang == other.lang and self.vtitle == other.vtitle
+
+    @staticmethod
+    def parse_api_params(version_params):
+        """
+        an api call contains ref and list of version_params
+        a version params is string divided by pipe as 'lang|vtitle'
+        this function takes the list of version_params and returns list of VersionsParams
+        """
+        version_params_list = []
+        for params_string in version_params:
+            lang, vtitle = split_query_param_and_add_defaults(params_string, 2, [''])
+            vtitle = vtitle.replace('_', ' ')
+            version_params = VersionsParams(lang, vtitle, params_string)
+            if version_params not in version_params_list:
+                version_params_list.append(version_params)
+        return version_params_list
+
```

Comment:
A function for parsing the version param itself should sit here under parse_api_params, but both of them should be in a class that encapsulates the API logic.